src/main/java/com/google/jenkins/plugins/credentials/oauth/RemotableGoogleCredentials.java

```diff
@@ -49,6 +50,14 @@ public RemotableGoogleCredentials(
       GoogleRobotCredentials credentials,
       GoogleOAuth2ScopeRequirement requirement,
       GoogleRobotCredentialsModule module) throws GeneralSecurityException {
+      this(credentials, requirement, module, false);
+  }
+
+  /* package */ RemotableGoogleCredentials(
+          GoogleRobotCredentials credentials,
+          GoogleOAuth2ScopeRequirement requirement,
+          GoogleRobotCredentialsModule module,
+          boolean mock) throws GeneralSecurityException {
```

Comment:
I don't like having explicit mock or testing options in the core code; the code should ideally be constructed in such a way as to make this unnecessary, and having mock options in security-related code in particular is concerning.

It appears that the only effect of this is to ignore the access token's expiration and the token string. It seems like that could be achieved by having test code construct its GoogleRobotCredentials object with appropriate expiration and string data. The tests are already using a FakeGoogleCredentials class, so they should be able to control that - is there a reason not to take that approach?